mob/src/main/java/com/github/phantazmnetwork/mob/spawner/BasicMobSpawner.java

```diff
@@ -3,64 +3,169 @@
 import com.github.phantazmnetwork.mob.MobModel;
 import com.github.phantazmnetwork.mob.MobStore;
 import com.github.phantazmnetwork.mob.PhantazmMob;
-import com.github.phantazmnetwork.mob.goal.Goal;
+import com.github.phantazmnetwork.mob.skill.Skill;
 import com.github.phantazmnetwork.neuron.bindings.minestom.entity.NeuralEntity;
 import com.github.phantazmnetwork.neuron.bindings.minestom.entity.Spawner;
 import com.github.phantazmnetwork.neuron.bindings.minestom.entity.goal.GoalGroup;
-import com.github.phantazmnetwork.neuron.bindings.minestom.entity.goal.NeuralGoal;
+import com.github.steanky.element.core.annotation.DependencySupplier;
+import com.github.steanky.element.core.context.ContextManager;
+import com.github.steanky.element.core.context.ElementContext;
+import com.github.steanky.element.core.dependency.DependencyModule;
+import com.github.steanky.element.core.dependency.DependencyProvider;
+import com.github.steanky.element.core.dependency.ModuleDependencyProvider;
+import com.github.steanky.element.core.key.KeyParser;
+import com.github.steanky.ethylene.core.ConfigElement;
+import com.github.steanky.ethylene.core.collection.ConfigNode;
+import com.github.steanky.ethylene.core.processor.ConfigProcessException;
+import com.github.steanky.ethylene.core.processor.ConfigProcessor;
+import it.unimi.dsi.fastutil.booleans.BooleanObjectPair;
 import it.unimi.dsi.fastutil.objects.Object2FloatArrayMap;
+import net.kyori.adventure.key.Key;
 import net.minestom.server.attribute.Attribute;
 import net.minestom.server.coordinate.Point;
+import net.minestom.server.entity.Entity;
 import net.minestom.server.entity.EquipmentSlot;
+import net.minestom.server.entity.metadata.EntityMeta;
 import net.minestom.server.instance.Instance;
 import net.minestom.server.item.ItemStack;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Map;
-import java.util.Objects;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Parameter;
+import java.util.*;
 
 /**
  * Basic implementation of a {@link MobSpawner}.
  */
 public class BasicMobSpawner implements MobSpawner {
 
+    private final Map<BooleanObjectPair<String>, ConfigProcessor<?>> processorMap;
+
     private final MobStore mobStore;
 
     private final Spawner neuralSpawner;
 
+    private final ContextManager contextManager;
+
+    private final KeyParser keyParser;
+
+    public static class Module implements DependencyModule {
+
+        private final NeuralEntity entity;
+
+        public Module(@NotNull NeuralEntity entity) {
+            this.entity = Objects.requireNonNull(entity, "entity");
+        }
+
+        @DependencySupplier("mob.entity.entity")
```

Comment:
it is fine to @Memoize here to prevent additional reflective calls since NeuralEntity will not change

Reply:
and i just mentioned some random account